File: apps/ios/RoundsLoyalty/Models/RewardInstance.swift
import Foundation

struct RewardInstance: Codable, Identifiable {
    let id: UUID
    let customerId: UUID
    let vendorId: UUID
    let rewardName: String
    let rewardDescription: String?
    let status: String // available, collection_requested, ready, collected, expired, cancelled
    let expiresAt: Date?
    let createdAt: Date
    var vendors: [Vendor]?
    var collections: [RewardCollection]?

    var vendor: Vendor? { vendors?.first }
    var collection: RewardCollection? { collections?.first }

    enum CodingKeys: String, CodingKey {
        case id
        case customerId = "customer_id"
        case vendorId = "vendor_id"
        case rewardName = "reward_name"
        case rewardDescription = "reward_description"
        case status
        case expiresAt = "expires_at"
        case createdAt = "created_at"
        case vendors
        case collections = "reward_collections"
    }
}

File: apps/android/app/src/main/java/com/roundsloyalty/app/data/models/RewardInstance.kt
package com.roundsloyalty.app.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class RewardInstance(
    val id: String,
    @SerialName("customer_id") val customerId: String,
    @SerialName("vendor_id") val vendorId: String,
    val status: String, // available, redeemed, expired
    @SerialName("reward_name") val rewardName: String,
    @SerialName("reward_description") val rewardDescription: String? = null,
    @SerialName("earned_at") val earnedAt: String? = null,
    @SerialName("redeemed_at") val redeemedAt: String? = null,
    @SerialName("expires_at") val expiresAt: String? = null,
)
